modules/core/src/smithy4s/http/HttpUnaryClientCodecs.scala

```diff
@@ -192,7 +197,10 @@ object HttpUnaryClientCodecs {
               case None => inputEncoders.fromSchema(endpoint.input, inputEncoderCache)
             }
 
-          val inputEncoder = (i: I) => F.map(baseRequest(endpoint))(inputWriter.write(_, i))
+          val prefixedInputWriter: HttpRequest.Writer[Blob, I] =
+            if (hostPrefixInjection) inputWriter.combine(HttpRequest.Writer.hostPrefix(endpoint)) else inputWriter
```

Comment:
a `combineIf` method could make this easier to digest:
```
val prefixedInputWriter: HttpRequest.Writer[Blob, I] = inputWriter.combineIf(hostPrefixInjection, HttpRequest.Writer.hostPrefix(endpoint))`

```

Reply:
it's not a bad idea but I don't think a single pain point warrants it. Let's avoid increasing the API surface for now